apps/condo/domains/miniapp/schema/B2BAccessToken.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Generated by `createschema miniapp.B2BAccessToken 'sessionId:Text; context:Relationship:B2BAppContext:CASCADE; rightSet:Relationship:B2BAppAccessRightSet:CASCADE; expiresAt:Text;'`
+ */
+
+const crypto = require('crypto')
+
+const cookieSignature = require('cookie-signature')
+const dayjs = require('dayjs')
+const get = require('lodash/get')
+const isNil = require('lodash/isNil')
+
+const conf = require('@open-condo/config')
+const { userIsAdmin } = require('@open-condo/keystone/access')
+const { GQLError, GQLErrorCode: { BAD_USER_INPUT } } = require('@open-condo/keystone/errors')
+const { historical, versioned, uuided, tracked, softDeleted, dvAndSender } = require('@open-condo/keystone/plugins')
+const { GQLListSchema, find, getById, getByCondition } = require('@open-condo/keystone/schema')
+const { setSession, destroySession } = require('@open-condo/keystone/session')
+
+const { encryptionManager } = require('@condo/domains/common/utils/encryption')
+const { makeSessionData } = require('@condo/domains/common/utils/session')
+const access = require('@condo/domains/miniapp/access/B2BAccessToken')
+const {
+    ACCESS_TOKEN_SESSION_ID_PREFIX,
+    ACCESS_TOKEN_CONTEXT_NOT_EXISTS,
+    ACCESS_TOKEN_RIGHT_SET_NOT_EXISTS,
+    ACCESS_TOKEN_WRONG_RIGHT_SET_TYPE,
+    ACCESS_TOKEN_CONTEXT_DOES_NOT_MATCH_RIGHT_SET,
+    ACCESS_TOKEN_SERVICE_USER_DOES_NOT_MATCH_CONTEXT,
+    ACCESS_TOKEN_NEGATIVE_TTL,
+    ACCESS_RIGHT_SET_SCOPED_TYPE,
+} = require('@condo/domains/miniapp/constants')
+const { SERVICE_USER_FIELD } = require('@condo/domains/miniapp/schema/fields/accessRight')
+const { getEnabledPermissions } = require('@condo/domains/miniapp/utils/permissions')
+
+const { COOKIE_SECRET } = conf
+
+async function getValidationError (item) {
+
+    const b2bAppContext = await getByCondition('B2BAppContext', {
+        id: item.context,
+        deletedAt: null,
+    })
+    if (!b2bAppContext) {
+        return ERRORS.CONTEXT_NOT_EXISTS
+    }
+    const accessRightSet = await getByCondition('B2BAppAccessRightSet', {
+        id: item.rightSet,
+        deletedAt: null,
+    })
+    if (!accessRightSet) {
+        return ERRORS.RIGHT_SET_NOT_EXISTS
+    }
+    if (accessRightSet.type !== ACCESS_RIGHT_SET_SCOPED_TYPE) {
+        return ERRORS.WRONG_RIGHT_SET_TYPE
+    }
+    if (accessRightSet.app && (accessRightSet.app !== b2bAppContext.app)) {
```

Comment:
logic is not exactly right, It potentially can be possible to have no accessRightSet.app (during set_null delete or something), so you should check that both context and rightSet has app field and they're matching

Reply:
Right now if accessRightSet.app === null, check will'be skipped

Reply:
Easiest way to fix it, add app condition in find where above, like 

```
await getByCondition('B2BAppAccessRightSet', {
        id: item.rightSet,
        deletedAt: null,
        app: { deletedAt: null }
        user: { deletedAt: null }
    })
```